### DISCUSSION TITLE
Invalid bytes preceding JSON output in PowerShell

### DISCUSSION THREAD

**Discussion by svsc49** [jotaen/klog#218] [Help / Question]
Hi. A very good and beautiful program and I enjoy working with it. Thank you for making it available free of charge.

However, I keep having a problem when creating the JSON using PowerShell. I use "| out-file intern.json" or "> intern.json". Then I open the JSON and Firefox gives me the answer that "SyntaxError: JSON.parse: unexpected character at line 1 column 1 of the JSON data" The first two characters are question marks in a triangle. If I manually copy the text from PowerShell and build a JSON with it, then it works.

+--- Screenshots ---+
![Unbenannt](https://user-images.githubusercontent.com/73939249/194416910-c054a47b-c68f-4086-b44c-e2cb2a02fe82.JPG)
![Unbenannt2](https://user-images.githubusercontent.com/73939249/194416914-1fb6b8aa-93cc-466e-9305-ca522a0da14e.JPG)
+--- German text below ---+

Hallo. Ein sehr gutes und schönes Programm und ich habe Freude damit zu arbeiten. Vielen Dank für die kostenfreie Zurverfügungstellung. 

Ich habe allerdings immer wieder ein Problem, beim Erstellen der JSON durch PowerShell. Dabei nutze ich "| out-file intern.json" oder "> intern.json". Dann öffne ich die JSON und Firefox gibt mir die Antwort, dass "SyntaxError: JSON.parse: unexpected character at line 1 column 1 of the JSON data" Die beiden ersten Zeichen sind Fragezeichen in einem Dreieck. Wenn ich den Text aus PowerShell händisch herauskopiere und damit eine JSON baue, dann funktioniert es.

**Comment by jotaen** [comment]
I think the problem is the [byte-order-mark (BOM) character](https://en.wikipedia.org/wiki/Byte_order_mark) that Windows Powershell prepends as first character when writing content into a file. That messes up the data for other programs that expect UTF-8 input, so therefore you see a scrambled character at the beginning of the text stream.

I’m not very familiar with Windows or Powershell, but maybe [this StackOverflow thread](https://stackoverflow.com/questions/5596982/using-powershell-to-write-a-file-in-utf-8-without-the-bom) is helpful to resolve the issue.
